catalyst/dl/callbacks/metrics/dice.py

```diff
@@ -33,4 +43,86 @@ def __init__(
         )
 
 
-__all__ = ["DiceCallback"]
+def calculate_dice(tp_fp_fn_dict: dict) -> np.array:
+    """Calculate list of Dice coefficients.
+    Args:
+        tp_fp_fn_dict: {"true_positives": true_positives,
+                        "false_positives": false_positives,
+                        "false_negatives": false_negatives}
+    Returns:
+    """
+    epsilon = 1e-7
+
+    true_positives = tp_fp_fn_dict["true_positives"]
+    false_positives = tp_fp_fn_dict["false_positives"]
+    false_negatives = tp_fp_fn_dict["false_negatives"]
+
+    dice = (2 * true_positives + epsilon) / (
+        2 * true_positives + false_positives + false_negatives + epsilon
+    )
+
+    if not np.all(dice <= 1):
+        raise ValueError("Dice index should be less than 1")
+
+    if not np.all(dice >= 0):
+        raise ValueError("Dice index should be more than 1")
+
+    return dice
+
+
+class MulticlassDiceMetricCallback(Callback):
+    def __init__(
+        self,
+        prefix: str = "dice",
+        input_key: str = "targets",
+        output_key: str = "logits",
+        class_names=None,
+        class_prefix="",
+        **metric_params
+    ):
+        super().__init__(CallbackOrder.Metric)
+        self.prefix = prefix
+        self.input_key = input_key
+        self.output_key = output_key
+        self.metric_params = metric_params
+        self.confusion_matrix = None
+        self.class_names = class_names  # dictionary {class_id: class_name}
```

Comment:
can it be just list, where indices are class_ids?

Reply:
I can imagine cases, where I want some classes to be excluded.